src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelSchema.scala

```diff
@@ -59,3 +43,41 @@ case class EntityMetadata(
 )
 
 case class EntityModel(schema : Map[String, EntityMetadata]) //entity name -> stuff about it
+
+object FlexibleModelSchema {
+  def memberTypeFromEntityType(entityType: String): Try[String] = {
+    val memberType = EntityTypes.types.get(entityType) match {
+      case Some(schema) => schema.memberType
+      case None => if (entityType.endsWith("_set"))
+        Some(entityType.replace("_set", "")) else
+        None
+    }
+    Try(memberType.get)
```

Comment:
With the fixed data model, it would be programmer error for the JSON to contain a set type without a `memberType` property. When the set type name is user supplied, we're enforcing that it must be named "${memberType}_set". Therefore, we need to report an error back to the user if they deviate from that.

I think something like this will do it:
```
Try(memberType.getOrElse(throw new RuntimeException("Entity set type name must end with \"_set\": " + entityType)))
```

Reply:
ugh. this doesn't work. it needs to return None instead of throwing an error because of how withMemberCollectionType in EntityClient is used. it makes this call regardless of whether is it a set or not. and None is the propper response when it is not a set. more suggestions welcome.

Reply:
Hmm... thinking out loud, haven't tried... but maybe `withMemberCollectionType`'s `op` parameter should be a `Try[Option[String]] => Future[PerRequestMessage]` so that the caller can decide what to do with the failure. When called from `EntityClient.importMembershipTSV`, throwing an error is right, I think... whereas the other two cases in `EntityClient` more-or-less expect a `None` result... I think... kinda hard to tell because they're passing it to functions that are called from other places that have different expectations about the value.